internal/service/streamprocessor/model.go

```diff
@@ -52,14 +53,21 @@ func NewStreamProcessorWithStats(ctx context.Context, projectID, instanceName st
 	if diags.HasError() {
 		return nil, diags
 	}
+	// we need to use pipelinePlan as the state must match the configuration in case the json has extra whitespace or different ordering
```

Comment:
I don't fully understand the reasoning here. What cases are you thinking here?

Reply:
A bit outdated, see comment below https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2519#discussion_r1728618273